storage/storage_dest.go

```diff
@@ -176,6 +180,7 @@ func (s *storageImageDestination) PutBlobWithOptions(ctx context.Context, stream
 
 	return info, s.queueOrCommit(*options.LayerIndex, addedLayerInfo{
 		digest:     info.Digest,
+		tocDigest:  options.TOCDigest,
```

Comment:
If this sets `tocDigest`, we can’t differentiate between non-chunked and chunked layer pulls; that again allows cross-format reuse we didn’t validate.